src/core/block_hash/starknet_block_hash.rs

```diff
@@ -20,21 +20,15 @@ pub fn calculate_tx_hashes_with_signatures(
     tx_hashes: Vec<Felt>,
     tx_signatures: Vec<Vec<Felt>>,
 ) -> Result<Vec<Felt>, SyscallHandlerError> {
-    let tx_and_signatures = zip(tx_hashes, tx_signatures).collect::<Vec<(Felt, Vec<Felt>)>>();
-    let mut hashes = Vec::with_capacity(tx_and_signatures.len());
-
-    for (hash, signature) in tx_and_signatures {
-        let hash = calculate_single_tx_hash_with_signature(hash, signature)?;
-        hashes.push(hash);
-    }
-
-    Ok(hashes)
+    zip(tx_hashes, tx_signatures)
+        .collect::<Vec<(Felt, Vec<Felt>)>>()
+        .into_iter()
```

Comment:
This seems redundant. Just map over the iterator.

Reply:
fixed!